exonum-node/src/consensus.rs

```diff
@@ -1035,19 +1090,29 @@ impl NodeHandler {
                     self.sign_message(TransactionsRequest::new(peer, txs))
                         .into()
                 }
+
                 RequestData::Prevotes(round, propose_hash) => self
                     .sign_message(PrevotesRequest::new(
                         peer,
-                        self.state.height(),
+                        self.state.epoch(),
                         round,
                         propose_hash,
                         self.state.known_prevotes(round, propose_hash),
                     ))
                     .into(),
+
                 RequestData::Block(height) => {
```

Comment:
What is the main difference between `Block` and `BlockOrEpoch` request?

Reply:
`Block` requests a block at the specified blockchain height; if the node does not have such a block, it ignores the request. `BlockOrEpoch`, on the other hand, requests a block at the specified blockchain height *or* a block skip with the specified or greater epoch.

- If the receiver has a block at the specified height, it returns this block.
- Otherwise, if the receiver has a block skip with the specified or greater epoch, it returns this block skip.
- Otherwise, the receiver does not reply.

I'll add this description to `BlockRequest` docs + other code comments.